[CI] MagnitudeSerializationTests testEqualsAndHashcode failing

**Build Scans:**
- [elasticsearch-periodic #11985 / release-tests](https://gradle-enterprise.elastic.co/s/tqcykenohtoca)
- [elasticsearch-periodic #11985 / release-tests](https://gradle-enterprise.elastic.co/s/rouh3n6q2mqzi)
- [elasticsearch-pull-request #127601 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/brtxncuzxc53g)
- [elasticsearch-pull-request #127566 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/nig673wc43cau)
- [elasticsearch-pull-request #127351 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/7kxzlg577mqe2)
- [elasticsearch-pull-request #127333 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/b6ak3tbjvafp4)
- [elasticsearch-pull-request #127291 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/fgbpkqcu4ksny)
- [elasticsearch-pull-request #127258 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/6jtn3g5etb3jg)
- [elasticsearch-pull-request #127211 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/vm3jtbn26m3de)
- [elasticsearch-pull-request #127202 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/d5hlmnwfe2l4a)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.expression.function.vector.MagnitudeSerializationTests.testEqualsAndHashcode" -Dtests.seed=59700DEF7E317431 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=uk -Dtests.timezone=Etc/GMT+4 -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.expression.function.vector.MagnitudeSerializationTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testEqualsAndHashcode'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Unknown NamedWriteable [org.elasticsearch.xpack.esql.core.expression.Expression][Magnitude]
```

**Issue Reasons:**
- [main] 2 consecutive failures in step release-tests
- [main] 14 consecutive failures in step checkpart3_release-tests
- [main] 16 failures in test testEqualsAndHashcode (1.8% fail rate in 885 executions)
- [main] 2 failures in step release-tests (100.0% fail rate in 2 executions)
- [main] 14 failures in step checkpart3_release-tests (100.0% fail rate in 14 executions)
- [main] 14 failures in pipeline elasticsearch-pull-request (2.4% fail rate in 591 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -436,15 +436,6 @@ tests:
   method: testMMRLimitedInput
   issue: https://github.com/elastic/elasticsearch/issues/144077
 - class: org.elasticsearch.xpack.esql.expression.function.vector.MagnitudeSerializationTests
-  method: testEqualsAndHashcode
-  issue: https://github.com/elastic/elasticsearch/issues/144078
-- class: org.elasticsearch.xpack.esql.expression.function.vector.MagnitudeSerializationTests
-  method: testSerialization
-  issue: https://github.com/elastic/elasticsearch/issues/144079
-- class: org.elasticsearch.xpack.esql.expression.function.vector.MagnitudeSerializationTests
-  method: testConcurrentSerialization
-  issue: https://github.com/elastic/elasticsearch/issues/144080
-- class: org.elasticsearch.xpack.esql.expression.function.vector.MagnitudeSerializationTests
   method: testConcurrentEquals
   issue: https://github.com/elastic/elasticsearch/issues/144081
 

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -435,9 +435,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.analysis.VerifierTests
   method: testMMRLimitedInput
   issue: https://github.com/elastic/elasticsearch/issues/144077
-- class: org.elasticsearch.xpack.esql.expression.function.vector.MagnitudeSerializationTests
-  method: testConcurrentEquals
-  issue: https://github.com/elastic/elasticsearch/issues/144081
 
 # Examples:
 #

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/vector/MagnitudeSerializationTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/vector/MagnitudeSerializationTests.java
@@ -7,11 +7,19 @@
 
 package org.elasticsearch.xpack.esql.expression.function.vector;
 
+import org.elasticsearch.xpack.esql.action.EsqlCapabilities;
 import org.elasticsearch.xpack.esql.core.expression.Expression;
 import org.elasticsearch.xpack.esql.core.tree.Source;
 import org.elasticsearch.xpack.esql.expression.AbstractUnaryScalarSerializationTests;
+import org.junit.Before;
 
 public class MagnitudeSerializationTests extends AbstractUnaryScalarSerializationTests<Magnitude> {
+
+    @Before
+    public void checkCapability() {
+        assumeTrue("v_magnitude available in snapshot", EsqlCapabilities.Cap.MAGNITUDE_SCALAR_VECTOR_FUNCTION.isEnabled());
+    }
+
     @Override
     protected Magnitude create(Source source, Expression child) {
         return new Magnitude(source, child);

